File: inst/Matlab/fdaM/ftpsite/temp/symsolve.m
function Xmat = symsolve(Asym, Bmat)
%SYMSOLVE solves ASYM Xmat = BMAT for Xmat where ASYM is symmetric
%  Last modified 24 April 2003

if max(max(abs(Asym-Asym')))/max(max(abs(Asym))) > 1e-10
    error('First argument is not symmetric.');
else
    Asym = (Asym + Asym')./2;
end
%  Choleski decomposition Asym = Rmat'*Rmat;
[Rmat,p] = chol(Asym);
if p > 0
    error('Argument ASYM is singular.');
end
temp = Rmat'\Bmat;
Xmat = Rmat\temp;

File: R/symsolve.R
symsolve <- function(Asym, Bmat)
{
  #  solves the system ASYM X = BMAT for X where ASYM is symmetric
  #  returns X
  n <- ncol(Asym)
  if (any(is.na(Asym))) stop("Asym has NA values.")
  if (max(abs(Asym-t(Asym)))/max(abs(Asym)) > 1e-10) {
    stop('Argument not symmetric.')
  } else {
    Asym <- (Asym + t(Asym))/2
  }
  Lmat <- chol(Asym)
  temp <- solve(t(Lmat),Bmat)
  Xmat <- backsolve(Lmat,temp)
  return(Xmat)
}
